code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm

```diff
@@ -0,0 +1,1553 @@
+	lifespan = 15
+	fade = 15
+	position = generator(GEN_SPHERE, 0, 17, UNIFORM_RAND)
+	velocity = list(0, 0.2)
+	scale = generator(GEN_NUM, 0.3, 0.5, UNIFORM_RAND)
+	grow = 0.05
+	spin = generator(GEN_NUM, 10, 20)
+
+/obj/effect/temp_visual/behemoth/earth_pillar
+	duration = 2 SECONDS
+	layer = ABOVE_ALL_MOB_LAYER
+
+/obj/effect/temp_visual/behemoth/earth_pillar/projectile
+	icon = 'icons/effects/128x128.dmi'
+	icon_state = "earth_pillar_exploded"
+	duration = 1.6 SECONDS
+	layer = ABOVE_ALL_MOB_LAYER
+	pixel_x = -48
+	pixel_y = -41
+
+/obj/effect/temp_visual/behemoth/earth_pillar/projectile/Initialize(mapload)
+	. = ..()
+	animate(src, time = duration - 0.5 SECONDS)
+	animate(alpha = 0, time = 0.5 SECONDS, easing = CIRCULAR_EASING|EASE_OUT)
+
+/obj/effect/temp_visual/behemoth/earth_pillar/destroyed
+	icon = 'icons/effects/128x128.dmi'
+	icon_state = "earth_pillar_destroyed"
+	duration = 1 SECONDS
+	pixel_x = -49
+	pixel_y = -38
+
+/obj/effect/temp_visual/behemoth/earth_pillar/destroyed/Initialize(mapload)
+	. = ..()
+	animate(src, time = 0.3 SECONDS)
+	animate(alpha = 0, time = 0.7 SECONDS, easing = CIRCULAR_EASING|EASE_OUT)
+
+/obj/effect/temp_visual/behemoth/earth_pillar/broken
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "earth_pillar_broken"
+	duration = 0.9 SECONDS
+	pixel_x = -32
+	pixel_y = -29
+
+/obj/effect/temp_visual/behemoth/earth_pillar/broken/Initialize(mapload)
+	. = ..()
+	animate(src, alpha = 0, time = duration, easing = CIRCULAR_EASING|EASE_OUT)
+
+/obj/structure/earth_pillar
+	name = "earth pillar"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "earth_pillar"
+	base_icon_state = "earth_pillar"
+	layer = ABOVE_LYING_MOB_LAYER
+	climbable = TRUE
+	climb_delay = 3 SECONDS
+	interaction_flags = INTERACT_CHECK_INCAPACITATED
+	density = TRUE
+	max_integrity = 300
+	soft_armor = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 0, BIO = 100, FIRE = 100, ACID = 0)
+	destroy_sound = 'sound/effects/behemoth/earth_pillar_destroyed.ogg'
+	coverage = 128
+	/// The owner of this object.
+	var/owner
+	/// Used for particles. Holds the particles instead of the mob. See particle_holder for documentation.
+	var/obj/effect/abstract/particle_holder/particle_holder
+	/// The amount of times a xeno needs to attack this to destroy it.
+	var/attacks_to_destroy = 3
+	/// The amount of times an Earth Pillar flashes before executing its interaction with Seismic Fracture.
+	var/warning_flashes = 3
+
+/obj/structure/earth_pillar/Initialize(mapload, new_owner)
+	. = ..()
+	owner = new_owner
+	playsound(src, 'sound/effects/behemoth/earth_pillar_rising.ogg', 40, TRUE)
+	particle_holder = new(src, /particles/earth_pillar)
+	particle_holder.pixel_y = -4
+	animate(particle_holder, pixel_y = 4, time = 1.0 SECONDS)
+	animate(alpha = 0, time = 0.6 SECONDS)
+	QDEL_NULL_IN(src, particle_holder, 1.6 SECONDS)
+	do_jitter_animation(jitter_loops = 5)
+
+/obj/structure/earth_pillar/Destroy()
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
```

Comment:
Why are you casting owner here instead of in the defenition on the object lol

Reply:
Fixed.